### PR TITLE
[Filter] Handle othercap only when it's valid. @open sesame 11/07 10:34

### LINKED ISSUE #549: [Investigate othercaps in tensor_filter] we need further investigation.
- [x] Add othercap logging to tensor_filter
- [ ] Find the values of othercap and why. --> Canceled. We do not have cases where it has valid value.
- [ ] Determine how to utilize othercap along with additional test cases / apps.

### PR DESCRIPTION
**Pull request by myungjoo**
Assume othercap is "ANY" if cap-size is ZERO.

Fixes #549

Signed-off-by: MyungJoo Ham <myungjoo.ham@samsung.com>

**Self evaluation:**
1. Build test: [x]Passed [ ]Failed [ ]Skipped
2. Run test: [x]Passed [ ]Failed [ ]Skipped

### CODE REVIEW

File: gst/tensor_filter/tensor_filter.c [hunk 1]
```diff
@@ -1067,7 +1073,7 @@ gst_tensor_filter_fixate_caps (GstBaseTransform * trans,
    */
   gst_caps_unref (othercaps);
```
[the remark this answers is not in the document]
Comment (myungjoo) [reply gst/tensor_filter/tensor_filter.c]: Oops! :)

File: gst/tensor_filter/tensor_filter.c [hunk 2]
```diff
@@ -1065,9 +1071,8 @@ gst_tensor_filter_fixate_caps (GstBaseTransform * trans,
   /**
    * To get the out-caps, GstTensorFilter has to parse tensor info from NN model.
    */
-  gst_caps_unref (othercaps);
 
-  result = gst_tensor_filter_transform_caps (trans, direction, caps, NULL);
+  result = gst_tensor_filter_transform_caps (trans, direction, caps, othercaps);
   result = gst_caps_make_writable (result);
```
Comment (jijoongmoon) [inline gst/tensor_filter/tensor_filter.c]: I wonder no need to uref othercaps? Cause it is unreferenced previously, but after this pr, it is not unreferenced.

Comment (myungjoo) [reply gst/tensor_filter/tensor_filter.c]: At ```gst_tensor_filter_transform_caps ()::1047``` of this file, it unrefs othercap if it's not NULL and valid. 

Anyway, we may need to unref it even if the size is 0 if not othercap is non-NULL.

Comment (myungjoo) [reply gst/tensor_filter/tensor_filter.c]: I'll fix it by removing unref from gst_tensor_filter_transform_caps and adding unref to gst_tensor_filter_fixate_caps

Comment (myungjoo) [reply gst/tensor_filter/tensor_filter.c]: Ah.. yes you're right. I'll add unref

**Pull request merged**